the project 'Add Order model' is ready for the code review

misterSokor opened the pull request.

misterSokor committed:
the project 'Add Order model' is ready for the code review

Elena-Bruyako commented on src/main/java/com/store/litflix/controller/OrderController.java:
```diff
@@ -0,0 +1,40 @@
+    )
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public OrderResponseDto placeOrder(@RequestBody @Valid OrderRequestDto orderRequest,
```
implement all endpoints

Elena-Bruyako commented on src/main/java/com/store/litflix/controller/OrderController.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    private Long extractUserId(Authentication authentication) {
+        return Long.parseLong(authentication.getName());
```
```suggestion
        User user = (User) authentication.getPrincipal();
        return user.getId();
```

Elena-Bruyako commented on src/main/java/com/store/litflix/dto/order/OrderRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
`@Data` already includes Getter and Setter
```suggestion
```

Elena-Bruyako commented on src/main/java/com/store/litflix/dto/order/OrderRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+@Setter
+@Data
+public class OrderRequestDto {
+    private String shippingAddress;
```
add validation

Elena-Bruyako commented on src/main/java/com/store/litflix/mapper/OrderMapper.java:
```diff
@@ -0,0 +1,15 @@
+@Mapper(config = MapperConfig.class, uses = { OrderItemMapper.class })
+public interface OrderMapper {
+    @Mapping(target = "userId", source = "user.id")
+    @Mapping(target = "orderItems", source = "orderItems")
```
```suggestion
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/Order.java:
```diff
@@ -0,0 +1,55 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id", nullable = false)
```
```suggestion
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/Order.java:
```diff
@@ -0,0 +1,55 @@
+    private User user;
+
+    @Enumerated(EnumType.STRING)
+    @NotNull
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/Order.java:
```diff
@@ -0,0 +1,55 @@
+    @NotNull
+    private Status status;
+
+    @NotNull
```
```suggestion
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/Order.java:
```diff
@@ -0,0 +1,55 @@
+    @Column(precision = 12, scale = 2, nullable = false)
+    private BigDecimal total;
+
+    @NotNull
+    @CreationTimestamp
```
```suggestion
    @Column(nullable = false, columnDefinition = "TIMESTAMP")
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/Order.java:
```diff
@@ -0,0 +1,55 @@
+    @CreationTimestamp
+    private LocalDateTime orderDate;
+
+    @NotNull
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/OrderItem.java:
```diff
@@ -0,0 +1,40 @@
+    @JoinColumn(name = "book_id", nullable = false)
+    private Book book;
+
+    @NotNull
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/com/store/litflix/model/OrderItem.java:
```diff
@@ -0,0 +1,40 @@
+    @NotNull
+    private int quantity;
+
+    @NotNull
```
```suggestion
```

Elena-Bruyako commented on src/main/java/com/store/litflix/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+        User user =
+                userRepository.findById(userId).orElseThrow(
+                        () -> new EntityNotFoundException("User not found: " + userId));
+        String shippingAddress = orderRequestDto.getShippingAddress();
```
find shopping cart
check if cart includes cartItems
if not - OrderProcessingException

Elena-Bruyako commented on src/main/java/com/store/litflix/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found for user: " + userId));
+        BigDecimal total = BigDecimal.ZERO;
+        for (CartItem ci : cart.getCartItems()) {
```
bad naming

Elena-Bruyako commented on src/main/java/com/store/litflix/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+                        "Shopping cart not found for user: " + userId));
+        BigDecimal total = BigDecimal.ZERO;
+        for (CartItem ci : cart.getCartItems()) {
+            OrderItem oi = new OrderItem();
```
bad naming

Elena-Bruyako commented on src/main/java/com/store/litflix/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+
+        order.setTotal(total.setScale(2, RoundingMode.HALF_UP));
+
+        Order saved = orderRepository.save(order);
+        cart.getCartItems().clear();
+        return orderMapper.toDto(saved);
```
```suggestion
        orderRepository.save(order);
        cart.getCartItems().clear();
        return orderMapper.toDto(order);
```

Elena-Bruyako reviewed (changes requested):
Good job! Let’s improve your solution

misterSokor committed:
what was fixed:
- OrderController: switched to extracting userId from Authentication principal
- OrderRequestDto: added validation, removed redundant @Getter/@Setter
- Order/OrderItem entities:
  - replaced @OneToOne with @ManyToOne for User relation
  - added @Column(nullable = false) for required fields
  - ensured @Enumerated(EnumType.STRING) and @CreationTimestamp include column definitions
- OrderMapper: cleaned redundant mappings, delegated OrderItem mapping properly
- OrderServiceImpl:
  - added shopping cart existence and non-empty validation
  - improved variable naming for clarity
  - fixed order saving/return flow to avoid redundant save

misterSokor committed:
add README.md

misterSokor committed:
updated README file

Elena-Bruyako commented on src/main/java/com/store/litflix/controller/OrderController.java:
```diff
@@ -0,0 +1,97 @@
+        return orderService.placeOrder(orderRequest, userId);
+    }
+
+    private Long extractUserId(Authentication authentication) {
```
private method should be placed after all public methods

Elena-Bruyako commented on src/main/java/com/store/litflix/dto/order/OrderRequestDto.java:
```diff
@@ -0,0 +1,13 @@
+    @NotBlank(message = "Shipping address is required")
+    @Size(max = 255, message = "Shipping address must not exceed 255 characters")
+    private String shippingAddress;
+    private String status;
```
```suggestion
```

Elena-Bruyako commented on src/main/java/com/store/litflix/service/impl/OrderServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+
+    @Override
+    public OrderResponseDto placeOrder(OrderRequestDto orderRequestDto, Long userId) {
+        User user =
```
divide methods into separate methods

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution

misterSokor committed:
- OrderController: moved private helper extractUserId() after all public methods
- OrderRequestDto: removed unused status field, added validation for shippingAddress
- OrderServiceImpl: split placeOrder logic into smaller private methods
